src/component/2d/ft/Contours.tsx

```diff
@@ -129,28 +132,33 @@ function ContoursInner({ spectra, displayerKey }: ContoursInnerProps) {
 
   return (
     <g clipPath={`url(#${displayerKey}clip-chart-2d)`} className="contours">
-      {spectra?.map((spectrum) => (
-        <g key={spectrum.id}>
-          {spectrum.display.isPositiveVisible && (
-            <ContoursPaths
-              id={spectrum.id}
-              sign="positive"
-              spectrum={spectrum}
-              color={spectrum.display.positiveColor}
-              onTimeout={timeoutHandler}
-            />
-          )}
-          {spectrum.display.isNegativeVisible && (
-            <ContoursPaths
-              id={spectrum.id}
-              sign="negative"
-              spectrum={spectrum}
-              color={spectrum.display.negativeColor}
-              onTimeout={timeoutHandler}
-            />
-          )}
-        </g>
-      ))}
+      {spectra?.map((spectrum) => {
+        const noise = calculateSanPlot('2D', (spectrum.data as NmrData2DFt).rr);
```

Comment:
@lpatiny Does it make sense to calculate the noise everytime a spectrum is rendered?